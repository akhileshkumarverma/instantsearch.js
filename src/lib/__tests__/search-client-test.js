import InstantSearch from '../InstantSearch';

describe('InstantSearch Search Client', () => {
  describe('Properties', () => {
    it('should have default `addAlgoliaAgent()` and `clearCache()` methods', () => {
      const search = new InstantSearch({
        indexName: '',
        searchClient: { search() {} },
      });

      expect(search.client.addAlgoliaAgent).toBeDefined();
      expect(search.client.clearCache).toBeDefined();
    });
  });

  describe('Lifecycle', () => {
    it('gets called on search', () => {
      const searchClientSpy = {
        search: jest.fn(),
      };

      const search = new InstantSearch({
        indexName: '',
        searchClient: searchClientSpy,
      });

      expect(searchClientSpy.search).not.toHaveBeenCalled();

      search.start();

      expect(search.helper.state.query).toBe('');
      expect(searchClientSpy.search).toHaveBeenCalledTimes(1);
    });

    it('calls the provided searchFunction when used', () => {
      const searchFunctionSpy = jest.fn(h => {
        h.setQuery('test').search();
      });

      const searchClientSpy = {
        search: jest.fn(),
      };

      const search = new InstantSearch({
        indexName: '',
        searchFunction: searchFunctionSpy,
        searchClient: searchClientSpy,
      });

      expect(searchFunctionSpy).not.toHaveBeenCalled();
      expect(searchClientSpy.search).not.toHaveBeenCalled();

      search.start();

      expect(searchFunctionSpy).toHaveBeenCalledTimes(1);
      expect(search.helper.state.query).toBe('test');
      expect(searchClientSpy.search).toHaveBeenCalledTimes(1);
      // expect(searchClientSpy.search).toHaveBeenCalledWith('test');
    });
  });
});
