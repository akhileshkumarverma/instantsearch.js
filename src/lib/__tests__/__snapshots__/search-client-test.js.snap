// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstantSearch Search Client Properties throws if no \`search()\` method 1`] = `"InstantSearch configuration error: \`searchClient\` must implement a \`search(requests)\` method."`;
